### DISCUSSION TITLE
Docker + Custom Server + HTTPS

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#40139] [Help]
Hello,

I am looking for feedback on what I am trying to currently do for a prod deployment.

A few of the things I am trying to do are as follows.
- Create a Docker image
- Run the next server using https, my assumption is this can only be done with a custom server.
- The application will be running behind an Apache reverse proxy with a `basePath`

I have created a docker image using the `with-docker` example, which works flawlessly, and I am able to serve it successfully behind the Apache reverse proxy with the correct base path.  I am able to accomplish this using `output: 'standalone'` in the `next.config.js`.  This seems to be working correctly, but now I am stuck with how to implement a custom server and `output: 'standalone'` at the same time.

It seems that this is what is required to get this to work correctly.  I really wish there was a way to avoid creating a custom server for the sole purpose of enabling HTTPS.  We will have HTTPS on the Apache server but I am being asked to implement " end to end " HTTPS. 

Is there an easier way to implement HTTPS on the next server out of the box or is it always going to just run a standard HTTP server without customizations?

**Comment by nagayev** [comment]
@sgreco-dev That's maybe helpful:
https://gist.github.com/cecilemuller/9492b848eb8fe46d462abeb26656c4f8

**Reply by ghost** [reply]
@nagayev I am not looking for how to set it up locally.  I need HTTPS for production, so I am trying to use the docker example as well as a custom server, but when using 'standalone' mode, it uses a server.js that it generates, mine does not work with standalone mode.

**Comment by larsqa** [comment]
Ever figured this out? This seems a bigger issue than expected: [#34599](https://github.com/vercel/next.js/discussions/34599)

**Reply by ghost** [reply]
I have not, I haven't gotten many replies in the places that I have asked these questions.  It seems odd to me that finding info on this is so difficult, I would think a common set up would be running a docker container and using HTTPS.
